### PR TITLE
James 2949 remove users file repository

### PR DESCRIPTION
**Pull request by rouazana**
Nice alternative to https://github.com/linagora/james-project/pull/2842 ^^

### CODE REVIEW

**Comment by chibenwa** [comment]
Changelog? Upgrade instructions?

**Comment by rouazana** [comment]
```
[5d0b808f0c3d731ed1bed7de51bac57d28e09fef] [ERROR]   DefaultMailboxesProvisioningFilterTest.createMailboxesIfNeededShouldNotGenerateExceptionsInConcurrentEnvironment:94 » Execution
```
fixed unstable test, test this please

**Comment by Arsnael** [comment]
```
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called? 
```
test this please

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
   - JDBCRecipientRewriteTable implementation. Please use JPARecipientRewriteTable instead.
   - JamesUsersJdbcRepository, DefaultUsersJdbcRepository. Please use JpaUsersRepository instead.
   - MailboxQuotaFixed matcher. Please use IsOverQuota instead.
+- UsersFileRepository, which was marked as deprecated for years
```
Comment (mbaechler) [inline CHANGELOG.md]: indent issue?

Comment (chibenwa) [reply CHANGELOG.md]: No actually this removal was not planned as part of 3.4.0 hence the level of indent is right.

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**